src/bindings/python/flux/future.py

```diff
@@ -185,6 +187,55 @@ def get(self):
     def incref(self):
         self.pimpl.flux_future_incref()
 
+    # asyncio functions
+
+    def __await__(self):
+        """
+        Await to work with asyncio.
+        """
+        if not hasattr(self, "_loop"):  # pylint: disable=attribute-defined-outside-init
+            import flux.asyncio  # pylint: disable=import-outside-toplevel
+            self._loop = flux.asyncio.loop  # pylint: disable=attribute-defined-outside-init
```

Comment:
Import should go at top level. Also, probably better to set the `_loop` attribute on the future when it is created rather than trying to look it up somehow and falling back to what might be the wrong loop.

Reply:
This is actually the convention I saw in most implementations of asyncio. We have to do it at this level because otherwise we get a circular import. Also pinging @trws for comment!